kubernetes/schema_job_spec.go

```diff
@@ -128,11 +131,17 @@ func jobSpecFields() map[string]*schema.Schema {
 			},
 		},
 		"ttl_seconds_after_finished": {
-			Type:         schema.TypeInt,
-			Optional:     true,
-			ForceNew:     true,
-			ValidateFunc: validateNonNegativeInteger,
-			Description:  "ttlSecondsAfterFinished limits the lifetime of a Job that has finished execution (either Complete or Failed). If this field is set, ttlSecondsAfterFinished after the Job finishes, it is eligible to be automatically deleted. When the Job is being deleted, its lifecycle guarantees (e.g. finalizers) will be honored. If this field is unset, the Job won't be automatically deleted. If this field is set to zero, the Job becomes eligible to be deleted immediately after it finishes.",
+			Type:     schema.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ValidateFunc: func(value interface{}, key string) ([]string, []error) {
+				v, err := strconv.Atoi(value.(string))
+				if err != nil {
```

Comment:
Although, then it will have a different failure (non-empty plan) but it's better to have the test fail than to be skipped every time. Then we can fix it some day.

Reply:
It fails because `TTLAfterFinished` is behind a feature gate that is disabled by default so it will always fail on clusters unless that feature gate has been enabled – and you can't enable it on GKE and so on.

Reply:
I realize the feature gate will prevent this specific field from taking effect on the kubernetes side, but the Kubernetes API has supported this field since 1.14 at least, which gives us the opportunity to test this attribute in our schema.

I think if we don't remove this feature gate skip logic, this test will never actually run, and I believe that's what's happening today. (GKE won't run it, and I know minikube was skipping it. I'm not sure if there is a platform we use that supports this feature gate).